contracts/src/proving/Logic.sol

```diff
@@ -35,6 +39,14 @@ library Logic {
         bytes32 verifyingKey;
     }
 
+    /// @notice A blob with a deletion criterion attached.
+    /// @param deletionCriterion The deletion criterion.
+    /// @param blob The bytes-encoded blob data.
+    struct ExpirableBlob {
+        DeletionCriterion deletionCriterion;
+        bytes blob;
+    }
+
```

Comment:
I would maybe move this struct definition in between the enum and the `Instance`.